internal/pkg/cli/pipeline_package.go

```diff
@@ -0,0 +1,224 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/aws/copilot-cli/internal/pkg/cli/list"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+)
+
+type packagePipelineVars struct {
+	name    string
+	appName string
+}
+
+type packagePipelineOpts struct {
+	packagePipelineVars
+
+	pipelineDeployer                pipelineDeployer
+	tmplWriter                      io.WriteCloser
+	ws                              wsPipelineReader
+	codestar                        codestar
+	store                           store
+	pipelineStackConfig             func(in *deploy.CreatePipelineInput) pipelineStackConfig
+	configureDeployedPipelineLister func() deployedPipelineLister
+	newSvcListCmd                   func(io.Writer, string) cmd
+	newJobListCmd                   func(io.Writer, string) cmd
+
+	//catched variables
+	pipelineMft *manifest.Pipeline
+	app         *config.Application
+	svcBuffer   *bytes.Buffer
+	jobBuffer   *bytes.Buffer
+}
+
+func (o *packagePipelineOpts) Execute() error {
+	pipelines, err := o.ws.ListPipelines()
+	if err != nil {
+		return fmt.Errorf("list all pipelines in the workspace: %w", err)
+	}
+
+	pipelinePath := ""
+	for _, pipeline := range pipelines {
+		if pipeline.Name == o.name {
+			pipelinePath = pipeline.Path
+			break
+		}
+	}
+	if pipelinePath == "" {
+		return fmt.Errorf("pipeline '%s' not found", o.name)
```

Comment:
nit: `%q` automatically applies the quotations!
```suggestion
		return fmt.Errorf("pipeline %q not found", o.name)
```